### ISSUE TITLE
Complex logic for conditionals

### ISSUE THREAD

**Issue by kethinov** [rooseveltframework/teddy#20]
Time for me to eat even more crow?

I know I said _"Teddy adheres to a mostly logic-less templates philosophy. `<if>` and `<unless>` statements in Teddy can only check for a single variable's presence or its precise value. They cannot evaluate complex logic; there are no ands, ors, xors, parentheticals, or math operators. Teddy is of the opinion that complex logic doesn't belong in templates. Evaluate such logic at the code level and pass the results to the templates through the model as readable variables."_ here: https://github.com/kethinov/teddy#notable-intentional-design-choices-and-limitations

However, on further reflection I think some exceptions to this are called for, as initially proposed here: https://github.com/rooseveltframework/teddy/issues/19

In addition to adding more boolean logic support, it might also be worth considering adding support for complex logic e.g. by adding a `cond` attribute that accepts a syntax like this:

``` html
<if cond='something <= somethingElse'>
{! do something if true !}
</if>
<if cond='something > somethingElse'>
{! do something if true !}
</if>
```

The main challenge here is in coming up with a list of rules around what syntax is valid in `cond`.

Some questions to answer:
- Which operators should be supported?
- Should boolean operators be supported from within `cond` even though that would overlap with the different implementation proposed here? https://github.com/rooseveltframework/teddy/issues/19
- Should parenthesis be supported?

**Comment by kethinov** [comment]
Blocked by https://github.com/kethinov/teddy/issues/32

**Comment by kethinov** [comment]
Closing as out of scope.
